vllm/entrypoints/openai/serving_chat.py

```diff
@@ -156,9 +156,13 @@ async def create_chat_completion(
 
         request_id = f"chat-{random_uuid()}"
         try:
-            guided_decode_logits_processor = (
-                await self._guided_decode_logits_processor(request, tokenizer))
-
+            guided_decoding_params = \
+                self._create_guided_decoding_params(request)
+            # Some requests for tools will use guided decoding
+            if (guided_json :=
+                    self._get_guided_json_from_tool(request)) is not None:
+                guided_decoding_params.guided_json = guided_json
```

Comment:
Would it make sense to move these calls into `(Chat)CompletionRequest.to_sampling_params`?

Reply:
Yeah, you right 👍